src/datasets/builder.py

```diff
@@ -1987,6 +1979,10 @@ def _download_and_prepare(self, dl_manager, verification_mode, **prepare_splits_
                 "`DirectRunner` (you may run out of memory). \nExample of usage: "
                 f"\n\t`{usage_example}`"
             )
+        if self._writer_batch_size is not None:
+            logger.warning(
+                "Writer batch size is not supported for beam pipelines yet. Using the default chunk size for writing."
+            )
```

Comment:
leaving a warning for the BeamBasedBuilder until we merge https://github.com/huggingface/datasets/pull/5364